starlette/middleware/errors.py

```diff
@@ -137,7 +137,7 @@ class ServerErrorMiddleware:
     def __init__(
         self,
         app: ASGIApp,
-        handler: typing.Optional[typing.Callable] = None,
+        handler: typing.Optional[typing.Callable[[Request, Exception], Response]] = None,
```

Comment:
This one is a bit hard to handle: `handler` can be either a sync callable or a coroutine:

```python
if is_async_callable(self.handler):
    response = await self.handler(request, exc)
else:
    response = await run_in_threadpool(self.handler, request, exc)
```

We could use `Callable[[Request, Exception], Response | Awaitable[Response]]`, and cast `handler` in the `if...else` above